### PR TITLE
Fix invoice check and selecting same card

### PR DESCRIPTION
**Pull request by alex-deaconu**
## Description
Fix invoice check and selecting same card

[stage-18]

## Motivation and Context
Billing frictions epic. Addresses feedback in #2381 

## How Has This Been Tested?
Tested changes locally; updated unit tests.

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
No

### CODE REVIEW

**Comment by alex-deaconu** [comment]
Changing from `ng-click` to `ng-change` did not work as expected when the user clicked and pressed cancel on the confirmation modal. Added a check for comparison of the payment sources instead.

Improved logic for the `isInvoiced` check. Checking for both `on` and `off` may be overkill but at this point we're not 100% sure of the api reliability in terms of that field, so it's better to be safe.

@ezequielc please review. Thanks!

File: web/scripts/billing/controllers/ctr-subscription.js [hunk 1]
```diff
@@ -27,8 +27,17 @@ angular.module('risevision.apps.billing.controllers')
       taxExemptionFactory.init();
 
       $scope.isInvoiced = function() {
-        return subscriptionFactory.item && subscriptionFactory.item.subscription &&
-          !subscriptionFactory.item.subscription.payment_source_id;
+        if (subscriptionFactory.item && subscriptionFactory.item.subscription &&
+          subscriptionFactory.item.subscription.auto_collection) {
+          return subscriptionFactory.item.subscription.auto_collection === 'off' ||
+            subscriptionFactory.item.subscription.auto_collection !== 'on';
```
Comment (ezequielc) [inline web/scripts/billing/controllers/ctr-subscription.js:33]: @alex-deaconu Isn't`auto_collection === 'off'` redundant if you check for `auto_collection !== 'on'`? Thanks

**Review by ezequielc — APPROVED** [review]
@alex-deaconu Looks good, one clarification inline. Thanks

File: web/scripts/billing/controllers/ctr-subscription.js (hunk 1 above)
Comment (alex-deaconu) [reply web/scripts/billing/controllers/ctr-subscription.js:33]: @ezequielc it is, I'm not sure why I felt compelled to check for both ;)

**Comment by alex-deaconu** [comment]
@ezequielc merging this, will fix the redundant check in the main branch. Thanks.

**Pull request merged**